keanu-project/src/main/java/io/improbable/keanu/tensor/intgr/Nd4jIntegerTensor.java

```diff
@@ -336,53 +336,42 @@ public IntegerTensor powInPlace(int exponent) {
 
     @Override
     public IntegerTensor minusInPlace(IntegerTensor that) {
-
         if (that.isScalar()) {
-            minusInPlace(that.scalar());
+            tensor.subi(that.scalar());
         } else {
-            INDArray indArray = unsafeGetNd4J(that);
-            tensor.subi(indArray);
+            INDArrayShim.subi(tensor, unsafeGetNd4J(that));
         }
-
         return this;
     }
 
     @Override
     public IntegerTensor plusInPlace(IntegerTensor that) {
-
         if (that.isScalar()) {
-            plusInPlace(that.scalar());
+            tensor.addi(that.scalar());
         } else {
-            INDArray indArray = unsafeGetNd4J(that);
-            tensor.addi(indArray);
+            INDArrayShim.addi(tensor, unsafeGetNd4J(that));
         }
-
         return this;
     }
 
     @Override
     public IntegerTensor timesInPlace(IntegerTensor that) {
-
         if (that.isScalar()) {
-            timesInPlace(that.scalar());
+            tensor.muli(that.scalar());
         } else {
-            INDArray indArray = unsafeGetNd4J(that);
-            tensor.muli(indArray);
+            INDArrayShim.muli(tensor, unsafeGetNd4J(that));
         }
-
         return this;
     }
 
     @Override
     public IntegerTensor divInPlace(IntegerTensor that) {
-
         if (that.isScalar()) {
-            divInPlace(that.scalar());
+            tensor.divi(that.scalar());
         } else {
-            INDArray indArray = unsafeGetNd4J(that);
-            Transforms.floor(tensor.divi(indArray), false);
+            INDArrayShim.divi(tensor, unsafeGetNd4J(that));
         }
-
+        Transforms.floor(tensor, false);
```

Comment:
is this a bug fix?

Reply:
Yup. If you divided by a scalar before you could get fractional numbers in the Integer buffer.